maestrowf/abstracts/interfaces/schedulerscriptadapter.py

```diff
@@ -112,62 +119,102 @@ def _substitute_parallel_command(self, step_cmd, nodes, procs):
         err_msg = "{} attempting to allocate {} {} for a parallel call with" \
                   " a maximum allocation of {}"
 
-        # We have three things that can happen.
-        # 1. We are partitioning the allocation into smaller chunks.
-        # We need to check for the launcher tag with specified values.
-        search = list(re.finditer(self.alloc_regex, step_cmd))
-        cmd = step_cmd
-        if search:
-            LOGGER.debug("Allocation setup found. cmd=%s", step_cmd)
-            for match in search:
-                # For each regex match that we found do:
-                # Collect the nodes and procs in the launch allocation.
-                alloc_nodes = match.group("nodes")
-                alloc_procs = match.group("procs")
-                # Compare the allocation to step allocation.
-                if int(alloc_nodes) > nodes:
-                    msg = err_msg.format(
-                        match.group(),
-                        alloc_nodes,
-                        "nodes",
-                        nodes
+        LOGGER.debug("nodes=%s; procs=%s", nodes, procs)
+        LOGGER.debug("step_cmd=%s", step_cmd)
+        # See if the command contains a launcher token in it.
+        alloc_search = list(re.finditer(self.launcher_regex, step_cmd))
+        if alloc_search:
+            # If we find that launcher nomenclature.
+            total_nodes = 0     # Total nodes we've allocated so far.
+            total_procs = 0     # Total processors we've allocated so far.
+            cmd = step_cmd      # The step command we'll substitute into.
+            for match in alloc_search:
+                LOGGER.debug("Found a match: %s", match.group())
+                _nodes = None
+                _procs = None
+                # Look for the allocation information in the match.
+                _alloc = match.group("alloc")
+                # Search for the legacy format.
+                _legacy = re.search(self.legacy_alloc, _alloc)
+                if _legacy:
+                    # nodes, procs legacy notation.
+                    _ = _alloc.split(",")
+                    _nodes = _[0]
+                    _procs = _[1]
+                    LOGGER.debug(
+                        "Legacy setup detected. (nodes=%s, procs=%s)",
+                        _nodes,
+                        _procs
                     )
-                    LOGGER.error(msg)
-                    raise ValueError(msg)
-                if int(alloc_procs) > procs:
-                    msg = err_msg.format(
-                        match.group(),
-                        alloc_procs,
-                        "procs",
-                        procs
+                else:
+                    # We're dealing with the new style.
+                    _nodes = re.search(self.node_alloc, _alloc)
```

Comment:
I think this allows you to have malformed allocation breakdown. The way I'm understanding the code, you would allow something like `[3n,3p,2n]` and you'd end up with 3 nodes and 3 processes currently. But, that's clearly wrong. This may imply that there is a need for a validation subcommand.